pkg/image/fetcher.go

```diff
@@ -35,6 +36,22 @@ type LayoutOption struct {
 	Sparse bool
 }
 
+type ImagePullChecker interface {
```

Comment:
I am not totally convinced about this name, I do not have any suggestion, but not sure. I will try to think about it

Reply:
@jjbustamante what about `ImagePullPolicyService` or `ImagePullPolicyHandler`

Reply:
I like ** ImagePullPolicyHandler**. Also, can you move the interface definition to the [client](https://github.com/buildpacks/pack/blob/main/pkg/client/client.go#L89) file? I think it must be along with the others

Reply:
@jjbustamante please one more question- should we need to create a mock of `NewPullPolicyManager` method for test cases (I think we should) - 
```
func NewPullPolicyManager(logger logging.Logger) ImagePullChecker {
	return &ImagePullPolicyManager{Logger: logger}
}
```
or can use it directly during tests.

Reply:
> I like ** ImagePullPolicyHandler**. Also, can you move the interface definition to the [client](https://github.com/buildpacks/pack/blob/main/pkg/client/client.go#L89) file? I think it must be along with the others

if we move the `ImagePullChecker` interface to the client file then there will be a import cycle between client and image package cause `Fetcher` struct has a field `imagePullChecker` type `ImagePullChecker`   interface.

Reply:
Oh! .. ok, let's keep it there for now, we will come back to that later